### PR TITLE
Add reddit parser

### PR DESCRIPTION
**Pull request by lachiejames**

### CODE REVIEW

File: src/reddit/parsers/data-parser.spec.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        expect(postData[1]).toEqual({
+            author: "FireBetAlwaysWorking",
+            dateCreated: new Date("2021-09-06T00:49:31.000Z"),
+            numAwards: 19,
+            numComments: 1005,
+            numUpvotes: 35795,
+            thumbnail: "https://b.thumbs.redditmedia.com/2Xi6BZTHiXqpPHZfD23NpKrgAHT28gm0P_hoPAq4QyQ.jpg",
+            title: "Yard sale find. The guy had no idea what he had.",
+        });
+    });
```
Comment (lachiejames) [inline src/reddit/parsers/data-parser.spec.ts:18]: Add test for empty list
